WordPress/gutenberg
# Site Editor: Add more template types

## What?
This adds more template types to the Site Editor template creation flow, specifically:
```
	'category',
	'author',
	'taxonomy',
	'date',
	'tag',
	'attachment',
```

## Why?
So that users can create a broader selection of templates. Addresses, in part, https://github.com/WordPress/gutenberg/issues/37407.

## How?
Adding more template types is trivial - we just add them to the array. This PR also adds icons to the menu for each of these to make it easier to scan as per https://github.com/WordPress/gutenberg/issues/37407. There are more changes suggested in https://github.com/WordPress/gutenberg/issues/37407, including altering the order of these items and dividing them into logical groups. This PR is the first step towards that goal.

I'm not 100% sure about all of these icons - I couldn't find a paperclip icon...

## Testing Instructions
0. Switch to emptytheme (or any theme without many templates)
0. Open the site editor
0. Click on "Templates"
1. Click on "Add New" (top right)
2. Confirm that you see the new template types in the drop down
3. Click on one of the new template types (e.g. Category) and create a new template
4. Open this template in the frontend of your site (e.g. /?cat=1) and confirm that you see the new template.

## Screenshots or screencast 
<img width="1440" alt="Screenshot 2022-03-11 at 10 08 21" src="https://user-images.githubusercontent.com/275961/157846730-3e8ab5c6-0972-406e-9416-1ee0553909fa.png">
